tests/mocha/serializer_test.js

```diff
@@ -0,0 +1,1757 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+// TODO: Move this into samples as part of the dev-tools package.
+/**
+ * Constructs a serializer test.
+ * @param {string} title The title of this testcase.
+ * @param {string} xml The XML to use for the round-trip test.
+ * @constructor
+ * @extends {TestCase}
+ */
+function SerializerTestCase(title, xml) {
+  this.title = title;
+  this.xml = xml;
+}
+SerializerTestCase.prototype = new testHelpers.TestCase();
+
+/**
+ * The XML we want to ensure round-trips through the serializer.
+ */
+SerializerTestCase.prototype.xml = '';
+
+
+/**
+ * Constructs a serializer test suite.
+ * @param {string} title The title of this test suite.
+ * @extends {TestSuite}
```

Comment:
`@extends {TestSuite<SerializerTestCase>}`

There is a also second template type defined on `TestCase` but I think it should be removed. I'll link you in the PR for discussion.